integration-tests/api/__tests__/admin/product.js

```diff
@@ -1737,6 +1737,129 @@ describe("/admin/products", () => {
       expect(variant).toEqual(undefined)
     })
 
+    it("successfully deletes a product variant and its associated option values", async () => {
+      const api = useApi()
+
+      // Validate that the option value exists
+      const optValPre = await dbConnection.manager.findOne(ProductOptionValue, {
+        variant_id: "test-variant-option_2",
+      })
+
+      expect(optValPre).not.toEqual(undefined)
+
+      // Soft delete the variant
+      const response = await api.delete(
+        "/admin/products/test-product/variants/test-variant_2",
+        {
+          headers: {
+            Authorization: "Bearer test_token",
+          },
+        }
+      )
+
+      expect(response.status).toEqual(200)
+
+      // Validate that the option value was deleted
+      const optValPost = await dbConnection.manager.findOne(
```

Comment:
Should we also add a find with deleted to ensure that it still exists with a deleted_at value in order that if someone remove the soft delete the test does not pass